📝 Add docs for log forwarding (#2599)
platformsh/platformsh-docs: docs/src/increase-observability/logs/forward-logs.md

   <div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">platform project:curl /integrations -X POST -d <span class="s1">'{ "type":"sumologic", "url":"{{< variable "COLLECTOR_URL" >}}", "category":"{{< variable "SOURCE_CATEGORY" >}}" }'</span></span></span></code></pre></div>

View your logs in the **Log Search** tab.

{{< /codetabs >}}

Logs start getting forwarded automatically after the integration has been resolved,
which can take up to an hour.

## Forward to a syslog endpoint

Syslog is a standard for message logging.
Many third-party services offer endpoints for ingesting syslog events.
You can forward your Platform.sh logs to any of those endpoints.

To enable this forwarding, run a command like the following:

```bash
platform project:curl /integrations -X POST --data '{ "type":"syslog", "host":"{{< variable "INGESTION_HOST" >}}", "port":"{{< variable "INGESTION_HOST_PORT" >}}" }'
```

`type`, `host`, and `port` are the only properties required for all endpoints.
The following table shows the other available properties:

| Property         | Type      | Default    | Description |

> | -------- --------| --------- | ---------- | ----------- |

Reviewer: 🐞 There is a display issue with the table in the output. I think it might be because of a missing hyphen.

> | -------- --------| --------- | ---------- | ----------- |

Reviewer: ```suggestion
| -----------------| --------- | ---------- | ----------- |
```

Author: ```suggestion
| ---------------- | --------- | ---------- | ----------- |
```

| `protocol`       | `string`  | `tls`      | The network protocol to use in the connection. Can be one of `tls`, `tcp`, or `udp`. Defaults to `tls`. |
| `facility`       | `string`  | `1` (user) | A [syslog facility code](https://en.wikipedia.org/wiki/Syslog#Facility) to attach with each log to identify the source. Can be a number from 0 to 23. |
| `message_format` | `string`  | `rfc5424`  | The standard to use for the message format. Can be `rfc5424` or `rfc3164`. |
| `auth_token`     | `string`  |            | The token to authenticate with the given service. |
| `auth_mode`      | `string`  |            | The mode for authentication with the given service. Can be `prefix` or `structured_data`. |
| `tls_verify`     | `boolean` | `true`     | Whether to verify Transport Layer Security (TLS) certification when using the TLS protocol. |

Place any property you want inside the `data` object of the cURL request.
This should let you connect to any service that has syslog endpoints.

> 

Reviewer: 😍 I think having the standard information in the table + the bulleted list with links to other docs works for now (cf conversation on Slack). We can change strategy later if we get feedback that suggests it's not helpful enough.

I understand that the bulleted list is pulled from another file, I just can't find exactly where I'm supposed to tell you that Datadog is the only link that is followed by a colon, so I'd suggest changing it to a hyphen to fit the general style of the list.

Author: You probably can't find the place because I deleted the list :grinning:

Author: It happens when I push changes after requesting a review :speak_no_evil:

---
title: Forward logs
description: Send your logs to a third-party service for further analysis.
observabilitySuite: true
---

You might use a service to analyze logs from various parts of your fleet.
You might want to consolidate all your logs in one place that everyone has access to
without needing to grant them access to each project individually.

In such cases, forward your logs from Platform.sh to a supported third-party service.

Log forwarding is available for Grid and {{% names/dedicated-gen-3 %}} projects.

> For {{% names/dedicated-gen-2 %}} projects, see how [log remotely with `rsyslog`](../../dedicated-gen-2/architecture/options.md#remote-logging).

Reviewer: see how **to** log remotely?

> For {{% names/dedicated-gen-2 %}} projects, see how [log remotely with `rsyslog`](../../dedicated-gen-2/architecture/options.md#remote-logging).

Author: ```suggestion
For {{% names/dedicated-gen-2 %}} projects, see how to [log remotely with `rsyslog`](../../dedicated-gen-2/architecture/options.md#remote-logging).
```

* [New Relic](https://newrelic.com/)
* [Splunk](https://www.splunk.com/)
* [Sumo Logic](https://www.sumologic.com/)

## Supported languages and services

The following table presents the languages and services that support logs in a format that can be forwarded:

| Languages                           | Services                             |
| ----------------------------------- | ------------------------------------ |
| {{< log-forwarding runtime=true >}} | {{< log-forwarding runtime=false >}} |

## Enable log forwarding

To enable log forwarding in a project, you need to be a [project admin](../../administration/users.md).

Enable log forwarding for a specific project by following the steps for your [selected third-party service](#supported-third-party-services).

{{< codetabs >}}
---
title=New Relic
highlight=false
file=none
---

1. Get an API key from New Relic.
   You need a [license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#license-key).
   Your New Relic organization likely has one, but you can create one by following the New Relic docs.
2. Choose an API endpoint.

>    You can use the US endpoint `https://log-api.newrelic.com/log/v1`

Author: ```suggestion
   You can use the U.S. endpoint `https://log-api.newrelic.com/log/v1`
```

   or the EU endpoint `https://log-api.eu.newrelic.com/log/v1`.
3. Create the integration with the following command:

   <div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">platform p:curl /integrations -X POST -d <span class="s1">'{ "type":"newrelic", "url":"{{< variable "API_ENDPOINT" >}}", "license_key": "{{% variable "LICENSE_KEY" %}}" }'</span></span></span></code></pre></div>

View your logs in the **Logs** dashboard.

<--->
---
title=Splunk
highlight=false
file=none
---

1. In Splunk, get an Event Collector token on [Splunk Platform](https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector#Create_an_Event_Collector_token_on_Splunk_Cloud_Platform)
   or [Splunk Enterprise](https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector#Create_an_Event_Collector_token_on_Splunk_Enterprise).

> 2. Determine your host. That is the Splunk instance that is collecting data.

Author: ```suggestion
2. Determine your host, which is the Splunk instance that's collecting data.
```
